### PR TITLE
chore: bump libp2p deps

### PR DESCRIPTION
**Pull request by wemeetagain**
**Motivation**

CI has been failing due to a bug in `@libp2p/tcp`. That bug should now be fixed in the latest version.

**Description**

- Update all libp2p deps and smooth out the yarn.lock
- Change test node names to be unique to allow better debugging
- Fix some tweaks for tests which is not required anymore.

### CODE REVIEW

**Comment by nflaig** [comment]
We should confirm that this resolves the active handles issue (related libp2p issues are closed)
- https://github.com/ChainSafe/lodestar/issues/5642

**Review by nazarhussain — DISMISSED** [review]

**Review by nazarhussain — DISMISSED** [review]

File: packages/beacon-node/src/network/core/networkCore.ts [hunk 1]
```diff
@@ -253,25 +253,28 @@ export class NetworkCore implements INetworkCore {
...
+      this.logger.debug("network reqResp closed");
+      this.attnetsService.close();
+      this.syncnetsService.close();
+      await this.libp2p.stop();
+      this.logger.debug("network lib2p closed");
+
+      this.closed = true;
+    } catch (err) {
+      this.logger.error("Error on NetworkCore.close()", {}, err as Error);
+    }
```
Comment (nflaig) [inline packages/beacon-node/src/network/core/networkCore.ts]: Why are we not throwing the error here, how is it ensured that everything is closed in that case?

There is an explicit exit in the beacon handler but this is only executed if there is an error
https://github.com/ChainSafe/lodestar/blob/e42d6cc6457051cf67ab7a3cdbfada25028d10f7/packages/cli/src/cmds/beacon/handler.ts#L141-L142

File: packages/beacon-node/test/unit/monitoring/service.test.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ import {MonitoringOptions} from "../../../src/monitoring/options.js";
 import {sleep} from "../../utils/sleep.js";
 import {startRemoteService, remoteServiceRoutes, remoteServiceError} from "./remoteService.js";
 
-describe("monitoring / service", () => {
+describe.skip("monitoring / service", () => {
```
Comment (nflaig) [inline packages/beacon-node/test/unit/monitoring/service.test.ts]: unrelated change

File: packages/beacon-node/test/unit/api/impl/validator/produceAttestationData.test.ts [hunk 3]
```diff
@@ -32,10 +32,11 @@ describe("api - validator - produceAttestationData", function () {
     };
   });
 
-  it("Should throw when node is not synced", async function () {
+  it.only("Should throw when node is not synced", async function () {
```
Comment (nflaig) [inline packages/beacon-node/test/unit/api/impl/validator/produceAttestationData.test.ts]: unrelated change

**Review by nflaig — CHANGES_REQUESTED** [review]
There are still a bunch of unrelated changes

File: packages/beacon-node/src/network/core/networkCore.ts (hunk 1 above)
Comment (nazarhussain) [reply packages/beacon-node/src/network/core/networkCore.ts]: This code execute when normal exit does not work. And `node` actually could not closed gracefully, that cause this issue. So not sure how else we can cleanup.

Comment (nflaig) [reply packages/beacon-node/src/network/core/networkCore.ts]: My point is just if you swallow the error here it won't execute the explicit handler, hence process could hang.

Note: the other process.exit(0) we have should ideally be removed if libp2p active handles issue is resolved

**Comment by wemeetagain** [comment]
file conflicts

File: packages/beacon-node/src/network/core/networkCore.ts (hunk 1 above)
Comment (nflaig) [reply packages/beacon-node/src/network/core/networkCore.ts]: I don't think this should be merged without a rationale / explanation for wrapping this in a try-catch and not propagating the errors up to caller.

- Did anything change when in libp2p that makes this necessary now?
- Why does our code not handle that correctly upstream?

IMO this try-catch might hang the process if there is an error in the network shutdown sequence (see above comments) and we should not hide it in this PR (bump libp2p) without having a clear reason for adding it.

File: packages/beacon-node/src/network/core/networkCore.ts [hunk 4]
```diff
@@ -253,7 +253,6 @@ export class NetworkCore implements INetworkCore {
   /** Destroy this instance. Can only be called once. */
   async close(): Promise<void> {
     if (this.closed) return;
-
```
Comment (nflaig) [inline packages/beacon-node/src/network/core/networkCore.ts]: unrelated diff

File: packages/beacon-node/test/e2e/network/gossipsub.test.ts [hunk 5]
```diff
@@ -19,7 +19,6 @@ describe("gossipsub / worker", function () {
 
 function runTests(this: Mocha.Suite, {useWorker}: {useWorker: boolean}): void {
   if (this.timeout() < 20 * 1000) this.timeout(150 * 1000);
-  this.retries(2); // This test fail sometimes, with a 5% rate.
```
Comment (nflaig) [inline packages/beacon-node/test/e2e/network/gossipsub.test.ts:22]: Why can this be removed now? Was this just related to libp2p issue?

Comment (nazarhussain) [reply packages/beacon-node/test/e2e/network/gossipsub.test.ts:22]: Because tests are performing more stable now, so we don't need to have retries here.

**Review by nflaig — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.12.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.12.0) :tada:
